Solaris/DmarcAggregateParser.php

```diff
@@ -80,6 +80,10 @@ function parse( $files ) {
 			}
 
 			try {
+                    		foreach (['dkim', 'spf'] as $type) {
```

Comment:
I think you've put this in the wrong place... it should be at line ~100 below. This block is just parsing the report headers, not individual records.